plugins/rebalance/rest.go

```diff
@@ -0,0 +1,279 @@
+	Volname           string
+	Status            RebalanceStatus
+	RebalanceID       uuid.UUID
+	RebalanceFiles    uint64
+	RebalanceData     uint64
+	LookedupFiles     uint64
+	RebalanceFailures uint64
+	RebalanceTime     uint64
+	SkippedFiles      uint64
+	CommitHash        uint64
+}
+
+func createRebalanceInfo(rebalance *RebalanceInfo) (*RebalanceInfo, error) {
+	v := new(RebalanceInfo)
+	v.Volname = rebalance.Volname
+	v.RebalanceID = uuid.NewRandom()
+	v.Status = rebalance.Status
+	v.CommitHash = rebalance.CommitHash
+	return v, nil
+}
+
+func rebalanceStartInfoResp(v *RebalanceInfo) *api.RebalanceInfo {
+
+	return &api.RebalanceInfo{
+		Volname:     v.Volname,
+		Status:      api.RebalanceStatus(v.Status),
+		RebalanceID: v.RebalanceID,
+	}
+}
+
+func registerRebalanceStartStepFuncs() {
+	transaction.RegisterStepFunc(StartRebalance, "rebal-start.Commit")
+}
+
+func registerRebalanceStopStepFuncs() {
+	transaction.RegisterStepFunc(StopRebalance, "rebal-stop.Commit")
+}
+
+func rebalanceStart(w http.ResponseWriter, r *http.Request) {
+
+	ctx := r.Context()
+	logger := gdctx.GetReqLogger(ctx)
+
+	//collect inputs from url
+	volname := mux.Vars(r)["volname"]
+
+	rebalance := new(RebalanceInfo)
+
+	//Check volname given
+	if volname == "" {
+		restutils.SendHTTPError(r.Context(), w, http.StatusBadRequest, "Volume not found", api.ErrCodeDefault)
+		return
+	}
+
+	//Validate rebalance command
+	vol, err := volume.GetVolume(volname)
+	if err != nil {
+		restutils.SendHTTPError(r.Context(), w, http.StatusNotFound, "Invalid volume", api.ErrCodeDefault)
+		return
+	}
+
+	if vol.State != volume.VolStarted {
+		restutils.SendHTTPError(r.Context(), w, http.StatusBadRequest, "Volume not started", api.ErrCodeDefault)
+		return
+	}
+
+	if vol.DistCount <= 1 {
+		restutils.SendHTTPError(r.Context(), w, http.StatusBadRequest, "Volume is not distribute type or contain only 1 brick",
+			api.ErrCodeDefault)
+		return
+	}
+
+	// TODO: TIER volume is not defined
+	/*Check volume is a tier volume or not
+	if vol.Type == volume.Tier {
+	        restutils.SendHTTPError(r.Context(), w, http.StatusBadRequest, "Rebalance operations are not supported on a tier volume",
+	                                api.ErrCodeDefault)
+	        return
+	}*/
+
+	// Check rebalance is already in process
+	if rebalance.Status == Inprogress {
+		restutils.SendHTTPError(r.Context(), w, http.StatusBadRequest, "Rebalance is already in progess",
+			api.ErrCodeDefault)
+		return
+	}
+
+	// Check for remove- brick pending
+	//TODO
+
+	//A simple transaction to start rebalance
+	txn := transaction.NewTxn(ctx)
+	defer txn.Cleanup()
+	lock, unlock, err := transaction.CreateLockSteps(volname)
+	if err != nil {
+		restutils.SendHTTPError(r.Context(), w, http.StatusInternalServerError, err.Error(), api.ErrCodeDefault)
+		return
+	}
+
+	txn.Nodes = vol.Nodes()
+	txn.Steps = []*transaction.Step{
+		lock,
+		{
+			DoFunc: "rebal-start.Commit",
+			Nodes:  []uuid.UUID{gdctx.MyUUID},
```

Comment:
This will start rebalance process only in current node. Use `Nodes: txn.Nodes` to start rebalance process in all Volume nodes